airflow/utils/json.py

```diff
@@ -81,6 +81,9 @@ def _default(obj):
         elif k8s is not None and isinstance(obj, (k8s.V1Pod, k8s.V1ResourceRequirements)):
             from airflow.kubernetes.pod_generator import PodGenerator
 
-            return PodGenerator.serialize_pod(obj)
+            try:
+                return PodGenerator.serialize_pod(obj)
+            except Exception:
```

Comment:
I think it makes sense to  log a warning here?

Reply:
sure thing